### PR TITLE
Fix error prediction in instrumental_predict

### PR DESCRIPTION
**Pull request by halflearned**
Solves #325

### CODE REVIEW

File: core/src/prediction/InstrumentalPredictionStrategy.cpp [hunk 1]
```diff
@@ -243,6 +243,6 @@ std::vector<double> InstrumentalPredictionStrategy::compute_debiased_error(
     error_bias += (residual_loto - residual) * (residual_loto - residual);
   }
   
-  error_bias *= ((num_trees - 1) / num_trees);
+  error_bias *= ((double)(num_trees - 1)) / (double)(num_trees);
```
Comment (jtibshirani) [inline core/src/prediction/InstrumentalPredictionStrategy.cpp]: I would just have one cast (two is redundant).

**Comment by halflearned** [comment]
Hey @jtibshirani, I added a couple of quick tests. It feels a bit hacky, because we're testing a quantity that was computed but not returned by the `compute_debiased_error` function. Do you have other ideas of what to test at the moment? After #288 we can have something more sensible.

File: core/src/prediction/InstrumentalPredictionStrategy.cpp (hunk 1 above)
Comment (halflearned) [reply core/src/prediction/InstrumentalPredictionStrategy.cpp]: Done.

**Comment by jtibshirani** [comment]
Thanks for adding those tests. I agree they're hacky -- it might have been nice to pull out a method `InstrumentalPredictionStrategy#compute_excess_error` method now and unit test it. However this is an important fix and I'm okay merging it with the understanding that we'll improve these tests as part of the work on #288.

**Pull request merged**